Sources/ComposableArchitecture/TestStore.swift

```diff
@@ -422,7 +422,7 @@ import XCTestDynamicOverlay
 /// [merowing.info]: https://www.merowing.info
 /// [exhaustive-testing-in-tca]: https://www.merowing.info/exhaustive-testing-in-tca/
 /// [Composable-Architecture-at-Scale]: https://vimeo.com/751173570
-public final class TestStore<State, Action> {
+public final class TestStore<State: Equatable, Action> {
```

Comment:
The `Equatable` constraint was only absent here when we supported `TestStore.scope`, but now it's effectively required everywhere, so we can just force the conformance up front here, which improves compiler errors, but also allows us to invoke an assertion helper in the `deinit`.